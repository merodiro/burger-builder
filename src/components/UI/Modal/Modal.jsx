import React, { Fragment } from 'react'

import classes from './Modal.module.css'
import Backdrop from '../Backdrop/Backdrop'

const Modal = ({ show, closed, children }) => {
  let modalClasses = [classes.Modal]
  show && modalClasses.push(classes.Opened)
  return (
    <Fragment>
      <Backdrop show={show} clicked={closed} />
      <div className={modalClasses.join(' ')}>{children}</div>
    </Fragment>
  )
}

export default Modal
