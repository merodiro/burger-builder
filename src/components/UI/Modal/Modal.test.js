import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import Modal from './Modal'

describe('<Modal>', () => {
  it('renders the same output', () => {
    const component = shallow(<Modal>Hello world</Modal>)

    expect(component).toMatchSnapshot()
  })

  it('renders with show property', () => {
    const component = shallow(<Modal show="true">Hello world</Modal>)

    expect(component).toMatchSnapshot()
  })
})
