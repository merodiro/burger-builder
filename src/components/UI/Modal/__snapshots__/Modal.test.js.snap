// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> renders the same output 1`] = `
<React.Fragment>
  <Backdrop />
  <div
    className="Modal"
  >
    Hello world
  </div>
</React.Fragment>
`;

exports[`<Modal> renders with show property 1`] = `
<React.Fragment>
  <Backdrop
    show="true"
  />
  <div
    className="Modal Opened"
  >
    Hello world
  </div>
</React.Fragment>
`;
